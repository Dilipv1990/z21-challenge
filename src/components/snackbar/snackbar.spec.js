import React from 'react';
import { shallow } from 'enzyme';
import CustomizedSnackbars from './snackbar';
import toJson from 'enzyme-to-json';
import Snackbar from '@material-ui/core/Snackbar';
import SnackbarContent from '@material-ui/core/SnackbarContent';

const props = {
  variant: 'error',
  message: 'err message',
  handleClose: jest.fn()
 };
 const wrapper = shallow(<CustomizedSnackbars {...props}/>);

describe('Customized Snackbar component test with Enzyme', () => {

   it('renders without crashing', () => {
     expect(toJson(wrapper)).toMatchSnapshot();
    });

  it('contains Snackbar component from material-ui', () => {
    const snackbar = wrapper.find(Snackbar);
    expect(snackbar.length).toBe(1);
    });

  it('contains SnackbarContent component from material-ui', () => {
    const snackbarContent = wrapper.find('MySnackbarContentWrapper');
    expect(snackbarContent.length).toBe(1);
    });
});