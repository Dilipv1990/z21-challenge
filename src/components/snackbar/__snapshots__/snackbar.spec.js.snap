// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customized Snackbar component test with Enzyme renders without crashing 1`] = `
<div>
  <WithStyles(ForwardRef(Snackbar))
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
    autoHideDuration={6000}
    onClose={[MockFunction]}
  >
    <MySnackbarContentWrapper
      message="err message"
      onClose={[MockFunction]}
      variant="error"
    />
  </WithStyles(ForwardRef(Snackbar))>
</div>
`;
